doc/MiniLanguageTutorial.md

```diff
@@ -132,51 +132,28 @@ Mini has the following types:
 
 > a type that has no values, so that no value of this type can ever exist
 
-## Equality and assignability for types
+## Equality and Assignability for types
 
-Two types are equal if they have the same structure. Type aliases, as defined by non-import declarations, do not create a new type but simply define a shorthand method for referring to the underlying type.  (For example, after the declaration "`type foo = uint`", foo and uint are the same type.)
+A value of type `V` is assignable to storage of type `S` if and only if any of the following are true:
 
-Every atomic type is equal to itself.
+* `S` and `V` are the same atomic type
+* `S` and `V` are options, and the inner type of `V` is assignable to the inner type of `S`
+* `S` and `V` are tuples with the same number of fields, and each field of `V` is assignable to the corresponding field of `S`
+* `S` and `V` are arrays, and the member type of `V` is assignable to the member type of `S`
+* `S` and `V` are fixed-size arrays with the same size, and the member type of `V` is assignable to the member type of `S`
+* `S` and `V` are structs with the same number of fields, and each field of `V` has the same name as and is assignable to that of `S`
+* `S` and `V` are maps, and the key and value types of `V` are correspondingly assignable to that of `S`
```

Comment:
Shouldn't maps be treated like a function from keyType to valueType, therefore map<K1, V1> is assignable to map<K2, V2> if K2 is assignable to K1 and V1 is assignable to V2?